### PR TITLE
fix(webhook): allow cspc validation based on supported compression algos

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

fixes:#237

### CODE REVIEW

**Review by shubham14bajpai — APPROVED** [review]
lgtm

File: pkg/webhook/cspc.go [hunk 1]
```diff
@@ -341,9 +341,13 @@ var (
 	// TODO: confirm all the compression types supported by control plane
 	// and update the map accordingly
 	SupportedCompression = map[string]bool{
-		"":    true,
-		"off": true,
-		"lz":  true,
+		"on":   true,
+		"off":  true,
+		"lzjb": true,
+		"gzip": true,
```
Comment (mittachaitu) [inline pkg/webhook/cspc.go:347]: We have to support values of `gzip-[1-9]` also since the data plane supports the same

Comment (mittachaitu) [reply pkg/webhook/cspc.go:347]: Command to create pool will be `zpool create <pool_name> <disk_path> -O compresstion=gzip-9`

File: pkg/webhook/cspc.go [hunk 2]
```diff
@@ -341,9 +341,13 @@ var (
 	// TODO: confirm all the compression types supported by control plane
 	// and update the map accordingly
 	SupportedCompression = map[string]bool{
```
Comment (mittachaitu) [inline pkg/webhook/cspc.go:343]: Add a comment saying this can be relooked when user request for gzip-[1-9] compression support...

**Review by mittachaitu — APPROVED** [review]
changes are good

File: pkg/webhook/cspc.go (hunk 2 above)
Comment (prateekpandey14) [reply pkg/webhook/cspc.go:343]: added the comment

File: pkg/webhook/cspc.go [hunk 3]
```diff
@@ -338,12 +338,16 @@ var (
 		cstor.PoolRaidz2:   isRaidz2BDCountValid,
 	}
 	// SupportedCompression is a map holding the supported compressions
-	// TODO: confirm all the compression types supported by control plane
-	// and update the map accordingly
+	// TODO: support to allow gzip-[1-9] compression algo can be added later if requested
```
Comment (mittachaitu) [inline pkg/webhook/cspc.go:341]:
```suggestion
	// TODO: support to allow gzip-[1-9] compression also can be added later if requested
```

**Review by mittachaitu — APPROVED** [review]
changes are good

File: pkg/webhook/cspc.go (hunk 3 above)
Comment (prateekpandey14) [reply pkg/webhook/cspc.go]: changed to  -- > `algorithm also`

**Pull request merged**